fastlane/lib/fastlane/actions/git_commit.rb

```diff
@@ -11,9 +11,9 @@ def self.run(params)
         skip_git_hooks = params[:skip_git_hooks] ? '--no-verify' : ''
 
         if params[:allow_nothing_to_commit]
-          nothing_staged = Actions.sh("git --no-pager diff --name-only --staged").empty?
-          UI.success("Nothing staged to commit ✅.") if nothing_staged
-          return if nothing_staged
+          repo_clean = Actions.sh("git status #{paths} --porcelain").empty?
```

Comment:
Can you please add a comment about this behaviour here, and at the description for "allow_nothing_to_commit" param, which "unstaged" changes are taken account and this way allows you to keep the repo dirty without failing?

We missed the behaviour in the last PR so it's definitely worth documenting. Ideally, we should cover it in unit tests but the way we test doesn't allow us to describe the exact situation for it due to `sh` being mocked or the difficulty to prepare a complete git repo just for a minor action. I can only come up leaving a comment as a way to avoid regressions for now.